ddtrace/tracer/remote_config.go

```diff
@@ -75,6 +76,27 @@ func (t *tags) toMap() *map[string]interface{} {
 	return &m
 }
 
+func (t *tracer) dynamicInstrumentationRCUpdate(u remoteconfig.ProductUpdate) map[string]state.ApplyStatus {
+
+	applyStatus := map[string]state.ApplyStatus{}
+
+	for k, v := range u {
+		log.Debug("Received dynamic instrumentation RC configuration for %s\n", k)
+		applyStatus[k] = state.ApplyStatus{State: state.ApplyStateUnknown}
+		passFullConfiguration(k, string(v))
+	}
+
+	// XXX: This function would typically respond with a status to say if the config was acknowledged
+	// but since we're extracting from BPF, we do not have a feedback loop and
+	return applyStatus
+}
+
+// passFullConfiguration is used as a stable interface to find the configuration in via bpf. Go-DI attaches
+// a bpf program to this function and extracts the raw bytes accordingly.
+//
+//go:noinline
+func passFullConfiguration(path, config string) {}
```

Comment:
@felixge I'm curious for your thoughts on this. `golint-ci` is complaining about these unused parameters. I can rename them each `_` and instrumentation seems to work just fine (can still get the parameter values) but perhaps there's other implications i'm unaware of. Renaming them removes entries for parameters in DWARF (though the bpf code for extracting them isn't going through DWARF). 

Is it possible in other architectures or in the future they wouldn't be passed at all and it'd be instrumentation? Would you recommend just overwriting the linter or changing the names to `_` ?

Reply:
Disassembly of the call to passFullConfiguration when the params are renamed to `_`:

```
  7dd7a8:       f9403be0        ldr     x0, [sp, #112]
  7dd7ac:       f94023e1        ldr     x1, [sp, #64]
  7dd7b0:       94000038        bl      7dd890 <gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer.passFullConfiguration>
 ```
 
I'm going to go ahead and make the change but still curious on your thoughts.

Reply:
Interesting question. I think the behavior you're relying on here is outside of the [language spec](https://go.dev/ref/spec), in particular:

* `//go:noinline` is not part of the language spec, and might break in the future.
* The calling convention for invoking passFullConfiguration() is [ABIInternal](https://go.googlesource.com/go/+/refs/heads/master/src/cmd/compile/abi-internal.md) which "is unstable and will change between Go versions"
* The reliance on the compiler not eliminating the call to `passFullConfiguration()` entirely.
* The reliance on the compiler to pass arguments when they are not used in the function body.
* The reliance on the compiler to pass arguments when they can't be used in the function body when they are being unnamed.

If I was you, I wouldn't be too worried about the last thing, at least not more worried than about the other things. The only way to make this stuff robust is to cover it with tests and run them against future versions of Go before they are being released.